site/app/libraries/database/DatabaseQueries.php

```diff
@@ -7122,20 +7163,29 @@ private function getUser($user_id, bool $is_numeric = false): ?User {
               FROM grading_registration
               GROUP BY user_id
             ) as sr ON u.user_id=sr.user_id
-            WHERE u.user_id=?",
-                [$user_id]
+            WHERE u.user_id IN {$placeholders}",
+                $user_id_list
             );
 
-            if ($this->course_db->getRowCount() > 0) {
-                $user = $this->course_db->row();
-                if (isset($user['grading_registration_sections'])) {
-                    $user['grading_registration_sections'] = $this->course_db->fromDatabaseToPHPArray($user['grading_registration_sections']);
+            $i = 0;
+            foreach ($details_list as &$details) {
+                if ($this->course_db->getRowCount() > 0) {
+                    $user = $this->course_db->rows()[$i];
+                    if (isset($user['grading_registration_sections'])) {
+                        $user['grading_registration_sections'] = $this->course_db->fromDatabaseToPHPArray($user['grading_registration_sections']);
+                    }
+                    $details = array_merge($details, $user);
                 }
-                $details = array_merge($details, $user);
+                $i++;
             }
         }
 
-        return new User($this->core, $details);
+        $users = [];
+        foreach ($details_list as &$details) {
```

Comment:
I don't think you need the & here as you make no edits to $details.
```suggestion
        foreach ($details_list as $details) {
```

Reply:
Yep, you're right.  That's a remnant from a previous attempt I made to fix this issue.  I'll delete it.